ddtrace/propagation/http.py

```diff
@@ -1008,6 +1011,47 @@ class HTTPPropagator(object):
     according to the propagation style set by ddtrace configurations.
     """
 
+    @staticmethod
+    def _get_sampled_injection_context(
+        trace_info: Union[Context, Span], non_active_span: Optional[Span] = None
+    ) -> Context:
+        """Handle sampling decision and return context for header injection.
+
+        If sampling_priority is already set, returns immediately. Otherwise, finds the
+        appropriate span and triggers sampling before returning the injection context.
+        """
+        # Extract context for header injection (non_active_span takes precedence)
+        injection_context = trace_info.context if isinstance(trace_info, Span) else trace_info
+
+        # Find root span for sampling decisions
+        if injection_context.sampling_priority is not None:
+            return injection_context
+        elif core.tracer is None:
+            # This should never happen, tracer should be initialized before headers can be injected.
+            log.error(
+                "No tracer found and injection context %s has no sampling priority, skipping sampling",
+                injection_context,
+            )
+            return injection_context
+
+        sampling_span: Optional[Span] = None
+        if non_active_span is not None:
+            # Deprecated: non_active_span takes precedence
+            sampling_span = non_active_span._local_root
+        elif isinstance(trace_info, Span):
+            # Use span's root for sampling
+            sampling_span = trace_info._local_root
+        elif (current_root := core.tracer.current_root_span()) and current_root.trace_id == trace_info.trace_id:
+            # Get the local root span for the current trace (if it is active, otherwise we can't sample)
+            sampling_span = current_root
+
+        # Sample the local root span before injecting headers.
+        if sampling_span:
```

Comment:
Should we add an else block here with a debug statement since it would show us if we were in the unusual case where we don't have a sampling_span but also never hit the earlier return condition?

Reply:
This behavior is tested in test_inject_context_and_span_different_trace_deprecated

Reply:
Since `non_active_span` is deprecated this shouldn't be a long term issue. We will kill this edge case